src/server/nbs/nbs_player.ts

```diff
@@ -0,0 +1,55 @@
+import * as bindings from 'bindings'
+import { EventEmitter } from 'events'
+
+const NBSPlayerAPI = bindings('nbs_player')
+
+export type NBSPacket = {
+  timestamp: number
+  hash: Buffer
+  payload: Buffer
+}
+
+export class NBSPlayer extends EventEmitter {
+
+  private player: any
+
+  constructor(private file: string) {
+    super()
+    this.player = new NBSPlayerAPI()
+    this.player.load(file, (timestamp?: number, hash?: Buffer, payload?: Buffer) => {
+      if (payload !== undefined) {
+        this.emit('packet', {
+          timestamp,
+          hash,
+          payload,
+        })
+      } else {
+        this.emit('end')
+      }
+    })
+  }
+
+  static of(opts: {file: string}) {
+    return new NBSPlayer(opts.file)
+  }
+
+  play() {
+    this.player.play()
+  }
+
+  pause() {
+    this.player.pause()
+  }
+
+  restart() {
+    this.player.restart()
+  }
+
+  step(steps: number) {
```

Comment:
Yeah, I added it because I know at some point in the future i'll want to step through individual nbs packets rather than play and pause.
When you run step, it'll stream that many packets without a delay.